File: backend/app/schemas/platform.py
from pydantic import BaseModel, Field


class Metric(BaseModel):
    label: str
    value: str
    hint: str = ""
    tone: str = "blue"


class WorkflowRunItem(BaseModel):
    id: str
    name: str
    target: str
    status: str
    progress: int
    updated_at: str


class CourseOperation(BaseModel):
    course: str
    class_name: str
    students: int
    completion: int
    alert: str


class ResourceJob(BaseModel):
    id: str
    name: str
    resource_type: str
    state: str
    time: str


class InsightCard(BaseModel):
    label: str
    value: str
    note: str


class TeacherWorkspace(BaseModel):
    stats: list[Metric]
    course_operations: list[CourseOperation]
    workflow_runs: list[WorkflowRunItem]
    resources: list[ResourceJob]
    insights: list[InsightCard]


class ResourceCreateRequest(BaseModel):
    name: str = Field(min_length=1, max_length=120)
    resource_type: str = Field(default="PDF", min_length=2, max_length=16)


class ResourceCreateResponse(BaseModel):
    resource: ResourceJob
    workflow_run: WorkflowRunItem


class ExerciseCreateRequest(BaseModel):
    knowledge_points: str = Field(min_length=1, max_length=120)
    difficulty: str = Field(default="balanced", min_length=1, max_length=32)
    count: int = Field(default=20, ge=1, le=100)


class ExerciseCreateResponse(BaseModel):
    task_id: str
    workflow_run: WorkflowRunItem
    title: str


class StudentCourse(BaseModel):
    id: str
    name: str
    teacher: str
    progress: int
    next_action: str
    mastery: str


class StudentTask(BaseModel):
    id: str
    title: str
    task_type: str
    due: str
    status: str


class MasteryPoint(BaseModel):
    label: str
    value: int


class ChatMessage(BaseModel):
    id: str
    role: str
    content: str
    time: str


class StudentWorkspace(BaseModel):
    stats: list[Metric]
    courses: list[StudentCourse]
    tasks: list[StudentTask]
    mastery_points: list[MasteryPoint]
    chat_messages: list[ChatMessage]
    recommendation: str


class ChatRequest(BaseModel):
    question: str = Field(min_length=1, max_length=500)
    course_id: str | None = None


class ChatResponse(BaseModel):
    answer: str
    messages: list[ChatMessage]


class TaskUpdateRequest(BaseModel):
    status: str = Field(min_length=1, max_length=20)


class ProviderStatus(BaseModel):
    id: str
    name: str
    provider_type: str
    status: str
    latency: str
    usage: int
    enabled: bool


class SystemModule(BaseModel):
    name: str
    owner: str
    state: str


class AuditEvent(BaseModel):
    actor: str
    action: str
    time: str


class AdminWorkspace(BaseModel):
    stats: list[Metric]
    providers: list[ProviderStatus]
    system_modules: list[SystemModule]
    audit_trail: list[AuditEvent]


class ProviderCreateRequest(BaseModel):
    name: str = Field(min_length=1, max_length=80)
    provider_type: str = Field(default="模型服务", min_length=1, max_length=40)
    latency: str = Field(default="260 ms", min_length=1, max_length=20)


class ProviderToggleResponse(BaseModel):
    provider: ProviderStatus

File: frontend/src/services/api/platform.ts
import { apiGet, apiPatch, apiPost } from "./client";

export type Metric = {
  label: string;
  value: string;
  hint?: string;
  tone: string;
};

export type WorkflowRunItem = {
  id: string;
  name: string;
  target: string;
  status: string;
  progress: number;
  updated_at: string;
};

export type CourseOperation = {
  course: string;
  class_name: string;
  students: number;
  completion: number;
  alert: string;
};

export type ResourceJob = {
  id: string;
  name: string;
  resource_type: string;
  state: string;
  time: string;
};

export type InsightCard = {
  label: string;
  value: string;
  note: string;
};

export type TeacherWorkspace = {
  stats: Metric[];
  course_operations: CourseOperation[];
  workflow_runs: WorkflowRunItem[];
  resources: ResourceJob[];
  insights: InsightCard[];
};

export type ResourceCreatePayload = {
  name: string;
  resource_type: string;
};

export type ResourceCreateResponse = {
  resource: ResourceJob;
  workflow_run: WorkflowRunItem;
};

export type ExerciseCreatePayload = {
  knowledge_points: string;
  difficulty: string;
  count: number;
};

export type ExerciseCreateResponse = {
  task_id: string;
  workflow_run: WorkflowRunItem;
  title: string;
};

export type StudentCourse = {
  id: string;
  name: string;
  teacher: string;
  progress: number;
  next_action: string;
  mastery: string;
};

export type StudentTask = {
  id: string;
  title: string;
  task_type: string;
  due: string;
  status: string;
};

export type MasteryPoint = {
  label: string;
  value: number;
};

export type ChatMessage = {
  id: string;
  role: string;
  content: string;
  time: string;
};

export type StudentWorkspace = {
  stats: Metric[];
  courses: StudentCourse[];
  tasks: StudentTask[];
  mastery_points: MasteryPoint[];
  chat_messages: ChatMessage[];
  recommendation: string;
};

export type ChatResponse = {
  answer: string;
  messages: ChatMessage[];
};

export type ProviderStatus = {
  id: string;
  name: string;
  provider_type: string;
  status: string;
  latency: string;
  usage: number;
  enabled: boolean;
};

export type SystemModule = {
  name: string;
  owner: string;
  state: string;
};

export type AuditEvent = {
  actor: string;
  action: string;
  time: string;
};

export type AdminWorkspace = {
  stats: Metric[];
  providers: ProviderStatus[];
  system_modules: SystemModule[];
  audit_trail: AuditEvent[];
};

export type ProviderCreatePayload = {
  name: string;
  provider_type: string;
  latency: string;
};

export function getTeacherWorkspace() {
  return apiGet<TeacherWorkspace>("/api/v1/teacher/workspace");
}

export function createResource(payload: ResourceCreatePayload) {
  return apiPost<ResourceCreateResponse>("/api/v1/resources", payload);
}

export function createExercise(payload: ExerciseCreatePayload) {
  return apiPost<ExerciseCreateResponse>("/api/v1/exercises", payload);
}

export function getStudentWorkspace() {
  return apiGet<StudentWorkspace>("/api/v1/student/workspace");
}

export function sendChatQuestion(question: string) {
  return apiPost<ChatResponse>("/api/v1/chat", { question });
}

export function updateStudentTask(taskId: string, status: string) {
  return apiPatch<StudentTask>(`/api/v1/student/tasks/${taskId}`, { status });
}

export function getAdminWorkspace() {
  return apiGet<AdminWorkspace>("/api/v1/admin/workspace");
}

export function createProvider(payload: ProviderCreatePayload) {
  return apiPost<ProviderStatus>("/api/v1/admin/providers", payload);
}

export function toggleProvider(providerId: string) {
  return apiPatch<{ provider: ProviderStatus }>(`/api/v1/admin/providers/${providerId}/toggle`);
}